cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -228,139 +258,202 @@ int embed_sim_init(gpi_sim_info_t *info)
         }
     }
 
-    PyObject *cocotb_module, *cocotb_init, *cocotb_args, *cocotb_retval;
-    PyObject *simlog_obj, *simlog_func;
-    PyObject *argv_list, *argc, *arg_dict, *arg_value;
+    PyObject *pCocotbModule, *pCocotbInit, *pCocotbArgs, *pInitReturn;
+    PyObject *pSimLogObject, *pLogFn, *pFilterFn;
+    PyObject *pArgvList, *pArgvItem;
 
-    cocotb_module = NULL;
-    arg_dict = NULL;
+    pCocotbModule = NULL;
+    pSimLogObject = NULL;
 
     // Ensure that the current thread is ready to call the Python C API
     PyGILState_STATE gstate = PyGILState_Ensure();
     to_python();
 
-    if (get_module_ref(COCOTB_MODULE, &cocotb_module))
+    if (get_module_ref(COCOTB_MODULE, &pCocotbModule))
         goto cleanup;
 
     // Obtain the loggpi logger object
-    simlog_obj = PyObject_GetAttrString(cocotb_module, "loggpi");
+    pSimLogObject = PyObject_GetAttrString(pCocotbModule, "loggpi");    // New reference
 
-    if (simlog_obj == NULL) {
+    if (pSimLogObject == NULL) {
         PyErr_Print();
         LOG_ERROR("Failed to get simlog object from loggpi\n");
+        goto cleanup;
     }
 
-    simlog_func = PyObject_GetAttrString(simlog_obj, "_logFromC");
-    if (simlog_func == NULL) {
+    // Obtain the function to use when logging from C code
+    pLogFn = PyObject_GetAttrString(pSimLogObject, "_logFromC");        // New reference
+    if (pLogFn == NULL) {
         PyErr_Print();
         LOG_ERROR("Failed to get the _logFromC method");
         goto cleanup;
     }
-
-    if (!PyCallable_Check(simlog_func)) {
-        PyErr_Print();
+    if (!PyCallable_Check(pLogFn)) {
+        if (PyErr_Occurred())
+            PyErr_Print();
```

Comment:
This is unreachable - from the docs,

> This function always succeeds.

Reply:
Ok, then the `PyErr_Print()` should also be removed, right? From the docs:

> void PyErr_PrintEx(int set_sys_last_vars)
> ...
>    Call this function only when the error indicator is set. Otherwise it will cause a fatal error!

Reply:
Yes - my comment refers to these two lines together.

Reply:
Removed all occurences. Resolving other comments.